### ISSUE TITLE
If I've segmented the target object

### ISSUE THREAD

**Issue by hwhyyds** [MS-Diffusion/MS-Diffusion#16]
If I've segmented the target object，how can I use the model output to retain the shape of the original object

**Comment by sherrydoge** [comment]
Do you mean that the shape of the object changes in pre-processing? In `inference.py`, we directly resize the input images. You can put the object in a square image without background information if you want to retain the shape.

**Comment by hwhyyds** [comment]
No, I mean that the shape of the target object has changed, such as the original image I used was bear cup, but the generated image is dog or any other cup

**Comment by hwhyyds** [comment]
Because I've already segmented the object, I want to diffuse the resulting image to contain the original image of the object

**Comment by sherrydoge** [comment]
Sorry, I am confused. Can you describe your input and the target output or give a specific example? Do you mean that you find MS-Diffusion fails to recreate your object?

**Comment by hwhyyds** [comment]
I gave the model a mug photo of a split bear ornament, but the picture the model generated was a mug photo of a duck ornament with other element. I want the model generate a mug photo of a split bear ornament.

**Comment by sherrydoge** [comment]
If the generated ornament is completely different from the input one, you should make sure that the bear ornament truly affects the model with the correct layout input (not zeros). If there is only some detail loss, you can try to increase the scale or add "bear" to the prompt.
